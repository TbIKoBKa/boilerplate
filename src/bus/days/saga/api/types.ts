// Types
import { Days } from '../../types';

export type FetchDaysContract = () => Promise<Days>
