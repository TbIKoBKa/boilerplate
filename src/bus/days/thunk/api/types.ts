// Types
import { Days } from '../../types';

export type FetchDays = () => Promise<Days>
