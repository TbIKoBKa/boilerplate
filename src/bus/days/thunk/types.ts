// Core
import { ActionReducerMapBuilder } from '@reduxjs/toolkit';

// Types
import * as types from '../types';

// Fetch
export const FETCH_WEATHER_ASYNC = 'days/FETCH_WEATHER_ASYNC';

// Contracts
export type FetchDaysAsyncContract = (builder: ActionReducerMapBuilder<types.Days>) => void;
