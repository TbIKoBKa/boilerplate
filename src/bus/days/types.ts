// Core
import { CaseReducer, PayloadAction } from '@reduxjs/toolkit';

// Types
export type WeatherType = 'sunny' | 'cloudy' | 'rainy'

// State
export type Day = {
    id: string
    rain_probability: number
    humidity: number
    day: number
    temperature: number
    type: WeatherType
}
export type Days = Day[]

// Contracts
export type SetWeatherContract = CaseReducer<Days, PayloadAction<Days>>
