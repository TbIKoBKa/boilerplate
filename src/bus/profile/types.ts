// Core
export type Profile = {
    firstName: string
    surname: string
    age: string | number
    email: string
    sex: string
    speciality: string
}

// Fill
export const FILL_PROFILE_DATA = 'FILL_PROFILE_DATA';
export type FillProfileDataActionType = {
    type: typeof FILL_PROFILE_DATA
    payload: Profile
}
export type FillProfileDataContract = (payload: Profile) => FillProfileDataActionType

export type ProfileActionTypes =
    | FillProfileDataActionType
