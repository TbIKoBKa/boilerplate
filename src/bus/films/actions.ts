// Types
import * as types from './types';

export const setFilmsAction: types.SetFilmsContract = (payload) => ({
    type: types.SET_FILMS,
    payload,
});
