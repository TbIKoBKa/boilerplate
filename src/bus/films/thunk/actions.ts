import * as types from './types';

export const fetchFilmsActionAsync: types.FetchFilmsContract = () => ({
    type: types.FETCH_FILMS_ASYNC,
});
