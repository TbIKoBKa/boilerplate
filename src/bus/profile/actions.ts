import * as types from './types';

export const fillProfileAction: types.FillProfileDataContract = (payload) => ({
    type: types.FILL_PROFILE_DATA,
    payload,
});
