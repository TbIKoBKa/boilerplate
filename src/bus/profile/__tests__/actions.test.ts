// Actions
import { fillProfileAction } from '../actions';

// Types
import { Profile } from '../types';

const testProfile: Profile = {
    firstName:  'Naruto',
    surname:    'Uzumaki',
    age:        18,
    email:      'narutolovesukraine@gmail.com',
    sex:        'male',
    speciality: 'Genin',
};

describe('profile actions', () => {
    test('should fill profile', () => {
        expect(fillProfileAction(testProfile)).toMatchSnapshot();
    });
});
