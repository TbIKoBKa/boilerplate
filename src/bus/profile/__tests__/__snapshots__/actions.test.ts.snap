// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions should fill profile 1`] = `
Object {
  "payload": Object {
    "age": 18,
    "email": "narutolovesukraine@gmail.com",
    "firstName": "Naruto",
    "sex": "male",
    "speciality": "Genin",
    "surname": "Uzumaki",
  },
  "type": "FILL_PROFILE_DATA",
}
`;
